cmd/common-methods.go

```diff
@@ -84,19 +84,23 @@ func getSourceStream(alias string, urlStr string, fetchStat bool) (reader io.Rea
 	if err != nil {
 		return nil, nil, err.Trace(alias, urlStr)
 	}
-	metadata = map[string][]string{}
+	metadata = map[string]string{}
 	if fetchStat {
 		st, err := sourceClnt.Stat(false)
 		if err != nil {
 			return nil, nil, err.Trace(alias, urlStr)
 		}
-		metadata = st.Metadata
+		for k, v := range st.Metadata {
+			if len(v) > 0 {
```

Comment:
HTTP header value can be empty, I see it sometimes in some HTTP responses but I am not sure how it can be useful. 

However, I think we don't have to bother ourselves to check if header value is empty or not at this stage. Any reason for this check ?

Reply:
guarding against errors since the value is a list, and the length could be 0?

Reply:
> guarding against errors since the value is a list, and the length could be 0?

yeah you are right, then I think we should do treat the case of len(v) == 0 and we set an empty string to metadata[k]. What do you think?